### PR TITLE
Api data for deposit website

### LINKED ISSUE #23: Public API: Expose data about public website
- [ ] Search by tracking number 
- [ ] Create test with rspec

### PR DESCRIPTION
**Pull request by Linzeur**
- Create namespace deposit within api
- Create controllers
- Add rspec
- Modify names of routes of namespace admin (not api)
- Fix views to change paths
- Close #23

### CODE REVIEW

File: app/controllers/api_controller.rb [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
   def authenticate_token
     authenticate_with_http_token do |token, _options|
       @current_user = User.find_by(authentication_token: token)
     end
   end
+
+  def render_unauthorized
+    errors = { errors: { message: 'Access denied' } }
+    render json: errors, status: :unauthorized
+  end
```
Comment (Linzeur) [inline app/controllers/api_controller.rb]: You could refactor this with method render_error_message

File: app/controllers/api/deposit/shipment_locations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    if params[:tracking_id]
+      shipment = Shipment.find_by(tracking_id: params[:tracking_id])
+      unless shipment.validate_stored?(current_user)
+        reception_date = DateTime.now
+        country = current_user.country
+        city = current_user.city
+        new_location = ShipmentLocation.create(shipment_id: shipment.id, reception_date: reception_date, country: country, city: city)
+        render json: new_location, status: :created
+      else
+        render_error_message("You can't store this shipment twice in row right here", :forbidden)
```
Comment (diegotc86) [inline app/controllers/api/deposit/shipment_locations_controller.rb]: Maybe this message will be clear: "The shipment is already stored in this location"

Comment (Linzeur) [reply app/controllers/api/deposit/shipment_locations_controller.rb]: Thanks, I am changing that right now!

File: app/controllers/api_controller.rb (hunk 1 above)
Comment (Linzeur) [reply app/controllers/api_controller.rb]: You're right! I will fix that

**Pull request merged**